pkg/query/logical/plan_indexscan.go

```diff
@@ -107,31 +107,49 @@ type indexScan struct {
 	traceState          series.TraceState
 }
 
-func (i *indexScan) Execute(ec executor2.ExecutionContext) ([]data.Entity, error) {
-	var chunkSet posting.List
-	for _, exprs := range i.conditionMap {
-		// TODO: Discuss which metadata should be used!
-		// 1) traceSeries Metadata: indirect mapping
-		// 2) indexRule Metadata: cannot uniquely determine traceSeries
-		// TODO: should pass correct shardID
-		chunks, err := ec.Search(*i.traceMetadata, 0, uint64(i.startTime), uint64(i.endTime), convertToConditions(exprs))
+func (i *indexScan) Execute(ec executor.ExecutionContext) ([]data.Entity, error) {
+	dataEntities := make([]data.Entity, 0)
+
+	// iterate over shards
+	for shardID := uint32(0); shardID < i.schema.ShardNumber(); shardID++ {
```

Comment:
Can we stop scanning when the limitation of results is reached?

Reply:
> Can we stop scanning when the limitation of results is reached?

I don't think so. If we want to do that,  we have to push down `Limit` as well as `Offset`.

Reply:
It's a performance issue more than a critical bug, we can improve it later. But we should mark it in the codes(TODO or something else) in case of missing it.

Reply:
> It's a performance issue more than a critical bug, we can improve it later. But we should mark it in the codes(TODO or something else) in case of missing it.

Sure. I've added a TODO there